docs/1.concepts/account-abstraction/chainsig.md

```diff
@@ -0,0 +1,114 @@
+:::
+
+#### Using Chain Signatures
+
+With Chain Signatures you can do the same but across many chains, for example Polygon:
+
+1. The developer creates a restricted NEAR account with a key
+2. The account is funded with `NEAR` and `MATIC`
+3. The user receives a key with limited control of the account
+4. The user uses the funded account to sign payloads calling controlled endpoints on Polygon
+5. The user returns the remaining funds to the developer and their account is unlocked
+
+This allows developers to pay for users to use arbitrary contracts on arbitrary chains.
+
+### Decentralized Clients
+
+A big problem in decentralized applications is that while the smart contracts are tamper-proof, the clients that access them generally are not. This allows practically complete control over any user account provided they are using the frontend assets that you serve. This has security, trust, and regulatory implications.
+
+When smart contracts can sign payloads you can start using [signed exchanges](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#name-introduction) (or polyfills) to require HTTP exchanges to be signed by a certain key. If it is not signed with this key the SSL certificate is considered invalid. This means that individual users cannot be served invalid frontends without it being generally observable and non repudiable.
+
+### Communication with private NEAR Shards
+
+Companies like [Calimero](https://www.calimero.network/) offer private NEAR shards. Currently, sending messages to and from these NEAR shards is troublesome. If each shard had the ability to sign their message queues, they could be securely sent from one shard to another. Thus you could communicate bidirectionally with any shard as easily as you can with a contract on your own shard.
+
+:::tip
+This could also simplify NEAR's sharding model, by treating each NEAR shard like one would a private shard.
+:::
+
+<!--
+## Future use cases
+
+### Chain decryptable messages [^1]
+
+Certain on chain keys allow the following flow:
+
+1. The sender sends encrypted data onto the chain
+2. When a receiver proves they have a right to the data it is re-encrypted with their key
+3. The receiver then decrypts the data
+
+This is useful for paid content, whatsapp messages and any private data on chain.
+
+Apart from the obvious this also enables:
+- **Secret-bid auctions with MEV protection** Users can submit bids that are encrypted at the end of the auction, the smart contract can decrypt all bids with a single evaluation.
+- **Dead man’s switch**. Journalists or whistleblowers can ensure that compromising information in their possession is automatically published if they were to become incapacitated
+- **One-time programs** Programs that can be executed only once on a single input, and that don’t leak anything about the program other than the result of the computation
+
+
+### On Chain Domain Control[^2]
```

Comment:
Probably best to drop this, I still haven't had the time to prove that this is actually feasible.

Reply:
Sure, these two were commented out. I'll remove them from the draft.